src/xr.js

```diff
@@ -75,6 +75,31 @@ function promise(args, fn) {
   )(fn);
 }
 
+function getURI(opts) {
+	let uri = opts.url;
+	let subPath = opts.subPath
+	if(!uri|| subPath.indexOf("://") !== -1) {
+		uri = subPath;
+		subPath = undefined;
+	}
+	if(subPath || opts.params) {
+		uri = uri.split('?')[0];
+	}
+	if(subPath) {
```

Comment:
This setup of `if` blocks seems dangerous to me... it creates a _lot_ of edge cases. :fearful: 

Also, the naming of the variables is throwing me off a little, it might make more sense to call the initial by `baseUrl` or the like, and then let url/uri be calculated based on what was passed